File: bindings/python/examples/how_tos/accounts_and_addresses/prepare_claim_outputs.py
import os

from dotenv import load_dotenv

from iota_sdk import ConsolidationParams, Utils, Wallet

# In this example we will claim all available outputs from an account.

# This example uses secrets in environment variables for simplicity which
# should not be done in production.
load_dotenv()

if 'STRONGHOLD_PASSWORD' not in os.environ:
    raise Exception('.env STRONGHOLD_PASSWORD is undefined, see .env.example')

wallet = Wallet(os.environ['WALLET_DB_PATH'])
wallet.set_stronghold_password(os.environ['STRONGHOLD_PASSWORD'])

account = wallet.get_account('Alice')

# Sync account to make sure account is updated with outputs from previous
# examples.
account.sync()
print('Account synced')

print('Preparing claim transaction...')
claimable = account.claimable_outputs('All')
prepared_transaction = account.prepare_claim_outputs(claimable)
print('Claim transaction:', prepared_transaction.prepared_transaction_data())
transaction = prepared_transaction.send()

# Wait for the consolidation transaction to get confirmed
block_id = account.retry_transaction_until_included(transaction.transactionId)

print(
    f'Transaction included: {os.environ["EXPLORER_URL"]}/block/{block_id}'
)

# Sync account
account.sync()
print('Account synced')

File: bindings/nodejs/examples/how_tos/advanced_transactions/prepare_claim_outputs.ts
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { OutputsToClaim, Wallet, initLogger } from '@iota/sdk';
require('dotenv').config({ path: '.env' });

// Run with command:
// yarn run-example ./how_tos/advanced_transactions/prepare_claim_outputs.ts

// This example claims all claimable outputs in the account, in a two-step process.
async function run() {
    initLogger();
    try {
        if (!process.env.STRONGHOLD_PASSWORD) {
            throw new Error(
                '.env STRONGHOLD_PASSWORD is undefined, see .env.example',
            );
        }

        const wallet = new Wallet({
            storagePath: process.env.WALLET_DB_PATH,
        });

        const account = await wallet.getAccount('Alice');

        await account.sync();

        // To sign a transaction we need to unlock stronghold.
        await wallet.setStrongholdPassword(process.env.STRONGHOLD_PASSWORD);

        const output_ids = await account.claimableOutputs(OutputsToClaim.All);
        console.log(`Available outputs to claim:`);
        for (const output_id of output_ids) {
            console.log(output_id);
        }

        const preparedTransaction = await account.prepareClaimOutputs(
            output_ids,
        );
        console.log(
            'Transaction prepared: ',
            preparedTransaction.preparedTransactionData(),
        );
        const transaction = await preparedTransaction.send();

        const blockId = await account.retryTransactionUntilIncluded(
            transaction.transactionId,
        );
        console.log(`Block sent: ${process.env.EXPLORER_URL}/block/${blockId}`);
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());
